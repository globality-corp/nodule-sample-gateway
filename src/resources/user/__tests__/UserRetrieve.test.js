import request from 'supertest';
import { Nodule } from '@globality/nodule-config';
import { signSymmetric } from '@globality/nodule-graphql';
import createApp from '../../../app';

let app;

const email = 'user@globality.com';
const token = signSymmetric({ email });


beforeEach(async () => {
    await Nodule.testing().fromObject({}).load();

    app = createApp();
});

it('should retrieve a single order', async () => {
    const query = `
        query {
          user {
            items {
                id
                email
            }
          }
        }
    `;

    const result = await request(app).post(
        '/gql/graphql',
    ).set(
        'Authorization', `Bearer ${token}`,
    ).send({
        query,
    });

    expect(result.body.errors).toBeUndefined();
    expect(result.body).toMatchSnapshot();
});
