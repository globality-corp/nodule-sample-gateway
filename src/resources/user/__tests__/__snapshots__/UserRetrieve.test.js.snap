// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should retrieve a single order 1`] = `
Object {
  "data": Object {
    "user": Object {
      "items": Array [
        Object {
          "email": "user@globality.com",
          "id": "5fc8f7a7-a590-4e96-adb6-e8dab8303408",
        },
      ],
    },
  },
}
`;
